### PR TITLE
Component

### PR DESCRIPTION
**Pull request by MelissaRoman06**
Guys, please check the code with the main functionalities working

### CODE REVIEW

File: src/main/java/Boiler.java [hunk 1]
```diff
@@ -1,4 +1,62 @@
...
+
+	/**
+	 * When the user pours water into the waterStrainer, waterStrainer capacityInUse
+	 * is set, also waterSensor status is changed.
+	 */
+	public void pourWater(int cupsOfWater) {
+		waterStrainer.setCapacityInUse(cupsOfWater);
+		waterSensor.setStatus(1);
+	}
+
```
Comment (aburgosu) [inline src/main/java/Boiler.java:24]: Add comment

File: src/main/java/Boiler.java [hunk 2]
```diff
@@ -1,4 +1,62 @@
...
+	 * First there has to be water in the boiler, this is why there is a condition
+	 * to get the waterSensor status, then the pressure valve should be closed and
+	 * then heatingElement is turned on.
+	 * 
+	 * @throws InterruptedException
+	 */
+	public void heatWater() throws InterruptedException {
+		if (waterSensor.getStatus() == 1) {
+			pressureValve.turnOff();
+
```
Comment (aburgosu) [inline src/main/java/Boiler.java]: delete unnecessary line jumps

File: src/main/java/CoffeeMaker.java [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+		lightIndicator = new LightIndicator();
+		boiler = new Boiler();
+		warmerPlate = new WarmerPlate();
+
+		System.out.println("Coffeemaker created");
+	}
+
+	public void startProcess(int qtyCups) throws InterruptedException {
+		warmerPlate.placePot();
+		//System.out.println("How many cups of coffee do you want?");
```
Comment (aburgosu) [inline src/main/java/CoffeeMaker.java]: delete comments

File: src/main/java/CoffeeMaker.java [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+	public CoffeeMaker() {
+		button = new Button();
+		lightIndicator = new LightIndicator();
+		boiler = new Boiler();
+		warmerPlate = new WarmerPlate();
+
+		System.out.println("Coffeemaker created");
+	}
+
```
Comment (aburgosu) [inline src/main/java/CoffeeMaker.java:21]: Add comments

File: src/main/java/CoffeeMaker.java [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+		if (button.isPressed()) {
+			System.out.println("startButton pressed");
+			startBrewing();
+			if (coffeeDripping(qtyCups)) {
+				lightIndicator.turnOn();
+				System.out.println("LightIndicator ON, coffee ready");
+			}
+		}
+	}
+
```
Comment (aburgosu) [inline src/main/java/CoffeeMaker.java:36]: add documentation comments

File: src/main/java/Component.java [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+
+public abstract class Component implements IComponent {
+	protected boolean status;
+
+	public Component() {
+		status = false;
+	}
+
+	public void turnOn() {
+
```
Comment (aburgosu) [inline src/main/java/Component.java]: delete unnecessary line jumps

File: src/main/java/Container.java [hunk 7]
```diff
@@ -0,0 +1,26 @@
+package main.java;
+
+public abstract class Container implements IContainer {
+	protected int capacityInUse;
+	protected int totalCapacity;
+
```
Comment (aburgosu) [inline src/main/java/Container.java:6]: add documentation comments

File: src/main/java/Main.java [hunk 8]
```diff
@@ -0,0 +1,12 @@
+package main.java;
+
+public class Main {
```
Comment (aburgosu) [inline src/main/java/Main.java]: Delete deprecated main method

File: src/main/java/PlateSensor.java [hunk 9]
```diff
@@ -1,5 +1,7 @@
-package java;
+package main.java;
 
-public class PlateSensor {
-   
+public class PlateSensor extends Sensor {
+	public PlateSensor() {
+		status = 0;
```
Comment (aburgosu) [inline src/main/java/PlateSensor.java]: what is the "0" status?
I recommend to use constants.

File: src/main/java/WarmerPlate.java [hunk 10]
```diff
@@ -1,5 +1,43 @@
...
+	ISensor plateSensor;
+	IComponent heatingElement;
+	IContainer pot;
 
+	WarmerPlate() {
+		plateSensor = new PlateSensor();
+		heatingElement = new HeatingElement();
+		pot = new Pot(12, 0);// CoffeeMaker's totalCapacity is 12 cups
+	}
+
```
Comment (aburgosu) [inline src/main/java/WarmerPlate.java:13]: add documentation comments

**Review by aburgosu — CHANGES_REQUESTED** [review]

File: src/main/java/Boiler.java (hunk 1 above)
Comment (MelissaRoman06) [reply src/main/java/Boiler.java:24]: Comments added

File: src/main/java/Boiler.java (hunk 2 above)
Comment (MelissaRoman06) [reply src/main/java/Boiler.java]: Line jumps deleted

File: src/main/java/CoffeeMaker.java (hunk 3 above)
Comment (MelissaRoman06) [reply src/main/java/CoffeeMaker.java]: Unnecessary comments deleted.

File: src/main/java/CoffeeMaker.java (hunk 4 above)
Comment (MelissaRoman06) [reply src/main/java/CoffeeMaker.java:21]: Comments added.

File: src/main/java/Component.java (hunk 6 above)
Comment (MelissaRoman06) [reply src/main/java/Component.java]: Line jumps deleted.

File: src/main/java/Main.java (hunk 8 above)
Comment (MelissaRoman06) [reply src/main/java/Main.java]: Main class deleted.

File: src/main/java/CoffeeMaker.java (hunk 5 above)
Comment (MelissaRoman06) [reply src/main/java/CoffeeMaker.java:36]: Comments added.

File: src/main/java/Container.java (hunk 7 above)
Comment (MelissaRoman06) [reply src/main/java/Container.java:6]: Comments added.

File: src/main/java/PlateSensor.java (hunk 9 above)
Comment (MelissaRoman06) [reply src/main/java/PlateSensor.java]: Constants implemented.

File: src/main/java/WarmerPlate.java (hunk 10 above)
Comment (MelissaRoman06) [reply src/main/java/WarmerPlate.java:13]: Comments added.

**Review by aburgosu — APPROVED** [review]

**Pull request merged**